tests/unit/virtualrouter/test_vr_plugin_handlers.py

```
@@ -0,0 +1,640 @@
+
+
+# ---------------------------------------------------------------------------
+# Dns — set_dns
+# ---------------------------------------------------------------------------
+@pytest.mark.virtualrouter
+class TestSetDns:
+    def test_set_dns_writes_resolv_conf(self):
+        plugin = dns_module.Dns.__new__(dns_module.Dns)
+        linux = sys.modules["zstacklib.utils.linux"]
+        linux.get_pid_by_process_name.return_value = "1234"
+        shell = sys.modules["zstacklib.utils.shell"]
```

Ruff (0.15.2)

[error] Local variable `shell` is assigned to but never used

Remove assignment to unused variable `shell`

(F841)